src/Domain/Mesh.hpp

```diff
@@ -123,6 +123,18 @@ class Mesh {
    */
   size_t number_of_grid_points() const noexcept { return extents_.product(); }
 
+  /*!
+   * \brief Returns the 1-dimensional index corresponding to the `Dim`
+   * dimensional `index`.
+   *
+   * The first dimension varies fastest.
+   *
+   * \see collapsed_index()
+   */
+  size_t storage_index(const Index<Dim>& index) const noexcept {
```

Comment:
Calling this member function would be equivalent to the existing `collapsed_index(index, mesh.extents())`, so why would you like to add it? If you want to have this wrapper, would a free function be better also here?

Reply:
Because it's useful and makes life easier. It also makes it so people will actually be aware it exists. I don't want it as a free function because I don't think it makes sense as a free function. The computation is strictly tied to a specific mesh and is part of that object.

Reply:
The other thing I want to make clear here is that this is the index used to store the data on the grid (hence storage_index, not collapsed_index). For the `Index` class this is not necessarily the relation ship, it is just a unique 1d index.